### PR TITLE
Add a skip button for challenge levels

### PR DESCRIPTION
**Pull request by balderdash**
Add a "Skip puzzle" button to challenge levels:

screenshot from 2017-08-14 14-51-26

Using this button (or the skip button in the dialog that pops up as soon as you open a challenge level) will send a milestone post with no program and a `testResult` of `SKIPPED`, which is -9.

### CODE REVIEW

File: apps/src/constants.js [hunk 1]
```diff
@@ -57,6 +57,7 @@ exports.TestResults = {
   LOCAL_FUNCTION_FAIL: -6,            // The program contains an unexpected JavaScript local function
   GENERIC_LINT_FAIL: -7,              // The program contains a lint error
   LOG_CONDITION_FAIL: -8,             // The program execution log did not pass a required condition
+  SKIPPED: -9,                        // Skipped, e.g. they used the skip button on a challenge level
```
Comment (joshlory) [inline apps/src/constants.js]: Worth checking with @ashercodeorg and @bencodeorg about this choice of constant?

Comment (ashercodeorg) [reply apps/src/constants.js]: Will there be problems as a result of the constant being defined in the same deploy as its usage (as a result of frontend mismatch)?

Comment (ashercodeorg) [reply apps/src/constants.js]: Given that we aggregate `UserLevel.best_result` to be the maximum, I'd think we might want to make this `-100` or some such (should it be greater than or less than `UNSUBMITTED_ATTEMPT`) to allow additional values greater than it.

File: lib/cdo/activity_constants.rb [hunk 2]
```diff
@@ -37,4 +38,9 @@ def self.finished?(result)
     return false if result.nil?
     result >= MINIMUM_FINISHED_RESULT
   end
+
+  def self.skipped?(result)
```
Comment (ashercodeorg) [inline lib/cdo/activity_constants.rb]: Please document the method.
```
# @param [Integer, nil] result The numerical score assigned to a user attempt.
# @return [Boolean] Whether the result indicates the attempt skipped the level, returning false if result is nil.

File: apps/src/constants.js (hunk 1 above)
Comment (balderdash) [reply apps/src/constants.js]: Yeah I'll split up the dashboard and apps parts and merge the dashboard part first. If you skip and hit an old frontend during the deploy, the starter code will be missing if you come back to the puzzle later.

Saved but unsubmitted progress should take precedence over a skip, I made it `-100` instead.

**Pull request merged**